zerver/decorator.py

```diff
@@ -335,10 +338,14 @@ def log_unsupported_webhook_event(request: HttpRequest, summary: str) -> None:
 
 def log_exception_to_webhook_logger(request: HttpRequest, err: Exception) -> None:
     extra = {"request": request}
+    # We deliberately skip logging this client error, as it results from a malformed request
+    # and doesn't indicate an issue on our end.
+    if isinstance(err, MissingHTTPEventHeaderError):
+        return
     # We intentionally omit the stack_info for these events, where
     # they are intentionally raised, and the stack_info between that
     # point and this one is not interesting.
-    if isinstance(err, AnomalousWebhookPayloadError):
+    elif isinstance(err, AnomalousWebhookPayloadError):
```

Comment:
`if` is actually better here -- you don't need `elif` if the previous block returned.